src/bookmarks/background/addition.js

```diff
@@ -52,13 +53,45 @@ export async function createNewPageForBookmark(id, bookmarkInfo) {
 
 export async function createBookmarkByExtension(url) {
```

Comment:
Does this function need to change anymore for this PR? It's only used in the existing overview bookmark code, right? If so, should revert all the changes to this

Reply:
No there is no need to change this function. Yes we are using this function in the bookmark from overview. I have just added the try cache block, nothing else so i think it should work.

Reply:
It does change the behaviour slightly though. L70 (indexing the `pageDoc`) can be reached even if `pageDoc` is still `undefined` (if `db.get` fails at L58). 
Also another thing to be aware of if you add error catching and don't throw it back to the caller: all calling code should be checked to make sure there isn't depending on that behaviour (not sure if problem here or not)